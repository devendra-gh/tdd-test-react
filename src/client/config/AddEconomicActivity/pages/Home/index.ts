import { IVariables } from '@tamm/app-composer';
import { PATH_HOME } from 'client/config/AddEconomicActivity/routes';
import functions from './functions';

const home = [
  {
    path: PATH_HOME, // path for router
    uniqueId: 'add-economic-activity.home', // uniqueId for caching and other purposes
    template: 'home', // template name, must be located in index of folder template/index
    title: 'addEconomicActivity.title', // title of the page, later it will be read from CMS
    init: functions.init,
    props: {
      title: 'addEconomicActivity.home.title',
      description: 'addEconomicActivity.home.description',
      showRelatedJourneyCard: true,
      startLogin: {
        title: 'addEconomicActivity.title',
        description: 'addEconomicActivity.home.startLogin.description',
        onClick: functions.onClick,
        buttonLabel: 'button.start',
      },
      process: {
        title: 'addEconomicActivity.home.process.title',
        steps: [
          {
            label: 'addEconomicActivity.addActivity',
            description: 'addEconomicActivity.addActivity.description',
          },
          {
            label: 'addEconomicActivity.getDEDApproval',
            description: 'addEconomicActivity.getDEDApproval.description',
          },
        ],
      },
      helpfulBlock: {
        callField: {
          onChange: () => {},
        },
        commentField: {
          onChange: () => {},
        },
        emailField: {
          onChange: () => {},
        },
        onChange: () => {},
        submitButton: {
          onClick: () => {},
        },
        telephoneField: {
          countries: [],
          onChange: () => {},
        },
      },
    },
    state: {
      mapState: [
        'stepsStatus',
        {
          steps: (state: IVariables) => {
            return [];
          },
        },
      ],
      mapDispatch: [
        'stepsStatus',
        'formData',
        'newActivityApiData',
        'helperData',
        'instanceId',
        'businessKey',
      ],
    },
  },
];

export default home;
