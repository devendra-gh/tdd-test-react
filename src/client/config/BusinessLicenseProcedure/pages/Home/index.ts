// import { REQUIRES_LOGIN } from '@tamm/app-composer';
import functions from './functions';

const home = [
  {
    path: ['/business-licence-procedure/'], // path for router
    uniqueId: 'licence-submit', // uniqueId for caching and other purposes
    template: 'businessLicenseProcedureHome', // template name, must be located in index of folder template/index
    title: 'businessLicenseProcedure.main.title', // title of the page, later it will be read from CMS
    props: {
      onStart: functions.onStart,
    },
    state: {
      mapState: [
        'loggedIn',
        'user',
        'instanceId',
        'businessKey',
        {
          steps: () => [],
        },
      ],
      mapDispatch: [
        'instanceId',
        'businessKey',
        'businessLicenseProcedureSteps',
      ],
    },
    // requires: [REQUIRES_LOGIN],
  },
];

export default home;
