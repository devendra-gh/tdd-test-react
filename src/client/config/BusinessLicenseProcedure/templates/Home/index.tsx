import React from 'react';
import Container from 'client/containers';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import JourneyLandingTemplate from '@tamm/ui-lib-v2-journey-landing-template';
import Sidebar from 'client/templates/Sidebar';
import { steps } from '../../steps';

/**
 * Home template
 * @param       {Object} props
 * @returns     {JSX}
 */
function Home(props: IVariables) {
  const { i18n } = props;
  return (
    <>
      <Container
        locale={props.locale}
        sidebar={
          <Sidebar
            currentStep={props.currentStep}
            currentSubStep={props.currentSubStep}
            i18n={props.i18n}
            steps={props.steps}
            showRelatedJourneyCard
            stepsStatus={props.stepsStatus}
          />
        }
      >
        <div className="ui-lib-story-container ui-lib-story-container_bg-light-grey">
          <JourneyLandingTemplate
            description={i18n('businessLicenseProcedure.home.description')}
            process={{
              steps: steps.map(step => ({
                description: i18n(`${step.name}.desc`),
                label: i18n(step.name),
              })),
              title: i18n('label.process'),
            }}
            startLogin={{
              buttonLabel: i18n('button.start'),
              description: i18n(
                'businessLicenseProcedure.home.start.description',
              ),
              title: i18n('businessLicenseProcedure.main.title'),
              onClick: () => props.onStart(props),
            }}
            title={i18n('label.serviceDescription')}
            helpfulBlock={{}}
          />
        </div>
        <div style={{ height: 200 }} />
      </Container>
    </>
  );
}

Home.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Home);
