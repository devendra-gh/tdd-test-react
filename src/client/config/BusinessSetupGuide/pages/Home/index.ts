import functions from './functions';

const home = [
  {
    path: ['/investment-compass/'], // path for router
    uniqueId: 'home-page', // uniqueId for caching and other purposes
    template: 'home', // template name, must be located in index of folder template/index
    title: 'bsg.home.title', // title of the page, later it will be read from CMS
    init: functions.init,
    onPageInit: functions.onPageInit,
    translations: {
      dictionaryId: 'buy-new-car',
      templateId: '4f3b7eee7185484f988ae2d898d666ba',
    },
    props: {
      breadcrumbs: [
        {
          label: 'bsg.breadcrumb.level0.title',
          link: 'https://www.tamm.abudhabi/',
        },
        {
          label: 'bsg.breadcrumb.level1.title',
          link: 'https://www.tamm.abudhabi/journeys/start-your-business',
        },
      ],
    },
    state: {
      mapState: [],
    },
  },
];

export default home;
