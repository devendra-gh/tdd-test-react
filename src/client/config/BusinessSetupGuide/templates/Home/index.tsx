import { IVariables, withTemplateHooks } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import Container from 'client/containers';
import React from 'react';
import { Link } from 'react-router-dom';
import { ICategory } from '../../data';
import './home.less';

/**
 * Home template
 * @param       {Object} props
 * @returns     {JSX}
 */
function Home(props: IVariables) {
  const { i18n, categoriesAr, categoriesEn } = props;
  const categories = props.locale === 'en' ? categoriesEn : categoriesAr;

  props.actions.breadcrumbs.update(props.breadcrumbs);

  return (
    <>
      <Container locale={props.locale}>
        <>
          <h3>{i18n('bsg.home.intro.title')}</h3>
          <p className="introText">{i18n('bsg.home.intro.text')}</p>
          <div className="categoryLinks">
            {categories.map((category: ICategory) => (
              <Link to={`/investment-compass/${category.link}`}>
                <div className="categoryLink">
                  <div className="iconWrapper">
                    <img
                      className="categoryIcon"
                      src={category.icon}
                      alt={category.title}
                    />
                  </div>
                  <br />
                  <p className="categoryTitle">{category.title}</p>
                </div>
              </Link>
            ))}
          </div>
        </>
        <div style={{ height: 200 }} />
        <div className="disclaimerHome">{i18n('bsg.disclaimer')}</div>
      </Container>
    </>
  );
}

Home.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Home);
