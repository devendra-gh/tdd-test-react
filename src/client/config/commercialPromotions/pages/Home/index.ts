// import { REQUIRES_LOGIN } from '@tamm/app-composer';
import { IVariables } from '@tamm/app-composer';
import functions from './functions';

const Home = [
  {
    path: '/commercial-promotions/home', // path for router
    uniqueId: 'commercial-promotions', // uniqueId for caching and other purposes
    template: 'service', // template name, must be located in index of folder template/index
    title: 'commericalPromotions.title', // title of the page, later it will be read from CMS,
    props: {
      startLogin: {
        description: 'commericalPromotions.title.description',
      },
      title: 'commericalPromotions.service',
      description: 'commericalPromotions.service.description',
      onSubmit: functions.onSubmit,
      showRelatedJourneyCard: true,
      helpfulBlock: {
        callField: {
          onChange: () => {},
        },
        commentField: {
          onChange: () => {},
        },
        emailField: {
          onChange: () => {},
        },
        onChange: () => {},
        submitButton: {
          onClick: () => {},
        },
        telephoneField: {
          countries: [],
        },
      },
    },

    state: {
      mapState: [
        'loggedIn',
        'stepsStatus',
        'history',
        {
          steps: (state: IVariables) => {
            return state.commercialPromotionsSteps;
          },
        },
      ],
      mapDispatch: ['stepsStatus', 'history'],
    },
    // requires: [REQUIRES_LOGIN],
  },
];

export default Home;
