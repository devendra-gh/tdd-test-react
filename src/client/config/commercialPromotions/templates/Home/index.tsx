import React from 'react';
import Sidebar from 'client/templates/Sidebar';
import JourneyLandingTemplate from '@tamm/ui-lib-v2-journey-landing-template';
import Container from 'client/containers';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
/**
 * ServiceHomeTemplate template
 * @param       {Object} props
 * @returns     {JSX}
 */

function ServiceHomeTemplate(props: IVariables) {
  const { i18n } = props;
  return (
    <>
      <Container
        locale={props.locale}
        sidebar={
          <Sidebar
            currentStep={props.currentStep}
            currentSubStep={props.currentSubStep}
            i18n={props.i18n}
            steps={props.steps}
            stepsStatus={props.stepsStatus}
            showRelatedJourneyCard={props.showRelatedJourneyCard}
          />
        }
      >
        <JourneyLandingTemplate
          startLogin={{
            title: i18n('commericalPromotions.title'),
            buttonLabel: i18n('button.start'),
            description: i18n(props.startLogin.description),
            onClick: () => {
              props.onSubmit(props);
            },
          }}
          helpfulBlock={props.helpfulBlock}
          title={props.i18n(props.title)}
          description={props.i18n(props.description)}
        />
        <div style={{ height: 200 }} />
      </Container>
    </>
  );
}

ServiceHomeTemplate.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(ServiceHomeTemplate);
