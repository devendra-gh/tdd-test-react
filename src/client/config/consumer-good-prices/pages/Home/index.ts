// import { REQUIRES_LOGIN } from '@tamm/app-composer';
import functions from './functions';

const home = [
  {
    path: ['/consumer-good-prices'], // path for router
    uniqueId: 'cgp-home', // uniqueId for caching and other purposes
    template: 'home', // template name, must be located in index of folder template/index
    title: 'main.title', // title of the page, later it will be read from CMS
    props: {
      onStart: functions.onStart,
      showTracker: false,
    },
    state: {
      mapState: ['loggedIn', 'user', 'steps', 'stepsStatus'],
      mapDispatch: ['steps', 'stepsStatus'],
    },
    // requires: [REQUIRES_LOGIN],
  },
];

export default home;
