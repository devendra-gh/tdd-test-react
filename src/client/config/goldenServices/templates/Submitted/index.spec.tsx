import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import '@testing-library/jest-dom/extend-expect';
import { Viewport, ViewportProvider } from '@tamm/ui-lib-v2-viewport';
import Home from './index';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: any) => component,
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('config/templates/Submitted', () => {
  let props: any;

  beforeEach(() => {
    props = {
      location: {},
      match: {},
      history: {
        push: jest.fn(),
      },
      locale: {
        switch: jest.fn(),
      },
      i18n: (key: string) => key,
      actions: {
        locale: {
          switch: jest.fn(),
        },
        breadcrumbs: {
          update: jest.fn(),
        },
        title: {
          update: jest.fn(),
        },
        hero: {
          update: jest.fn(),
        },
      },
      loggedIn: false,
    };
  });

  afterEach(cleanup);

  test('renders page', async () => {
    props.loggedIn = true;
    const { container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <Home {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
