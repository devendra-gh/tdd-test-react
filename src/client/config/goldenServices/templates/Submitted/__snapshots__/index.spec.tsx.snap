// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config/templates/Submitted renders page 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <div
    class="container"
  >
    <div
      class="ui-lib-notice"
    >
      <svg
        class="ui-lib-notice__icon ui-lib-notice__icon_success ui-lib-icon"
        data-key=""
        height="24px"
        tabindex="0"
        version="1.1"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#000000"
            transform="translate(-439.000000, -741.000000)"
          >
            <g
              transform="translate(439.000000, 741.000000)"
            >
              <path
                d="M12,-1.77635684e-15 C5.372583,-1.77635684e-15 0,5.372583 0,12 C0,18.627417 5.372583,24 12,24 C18.627417,24 24,18.627417 24,12 C23.9922852,5.37578119 18.6242188,0.0077147623 12,-1.77635684e-15 Z M18.927,8.2 L12.082,17.489 C11.9187536,17.7055896 11.6753283,17.8475051 11.4064219,17.8828578 C11.1375155,17.9182105 10.8656811,17.8440355 10.652,17.677 L5.764,13.769 C5.33266561,13.423822 5.26282203,12.7943344 5.608,12.363 C5.95317797,11.9316656 6.58266561,11.861822 7.014,12.207 L11.09,15.468 L17.317,7.017 C17.521192,6.71059002 17.8772511,6.54118265 18.2438115,6.57603752 C18.610372,6.61089239 18.9281223,6.84436978 19.0709064,7.18377241 C19.2136904,7.52317505 19.158405,7.91358568 18.927,8.2 Z"
              />
            </g>
          </g>
        </g>
      </svg>
      <h2
        class="ui-lib-notice__title"
      >
        goldenServices.subTitle.submitted
      </h2>
      <div
        class="ui-lib-notice__content"
      >
        <div>
          <p>
            goldenServices.subTitle.desc.submitted
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
